sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcSQLQueryBuilder.scala

```diff
@@ -151,6 +156,14 @@ class JdbcSQLQueryBuilder(dialect: JdbcDialect, options: JDBCOptions) {
     this
   }
 
+  /**
+   * Saves the hint value used to construct HINT clause.
+   */
+  def withHint(hint: String): JdbcSQLQueryBuilder = {
+    hintClause = dialect.getHint(hint)
+    this
+  }
+
```

Comment:
The with...() is used to data source pushdown API.
Please do not add the function.

Reply:
Maybe I should call `dialect.getHint()` directly in the `build()` method of `JdbcSQLQueryBuilder` and subclasses of `JdbcSQLQueryBuilder`?

Reply:
And those JdbcDialects that support hint overrides the `getHint` method so that the `getHint` method throws an exception by default?

Reply:
``` 
protected val hintClause: String = { 
    if (options.hint != "") {
      if (dialect.supportsHint) {
        options.hint
      } else {
        throw QueryCompilationErrors.hintUnsupportedForJdbcDatabasesError(this.getClass.getSimpleName)
      }
    } else {
      options.hint
    }
  }
```
How about code like this？